lib/widgets/routines/log.dart

```diff
@@ -197,3 +207,114 @@ class DayLogWidget extends StatelessWidget {
     );
   }
 }
+void showEditSessionDialog(BuildContext context, WorkoutSession session, RoutinesProvider routinesProvider) {
+  final _formKey = GlobalKey<FormState>();
+  final notesController = TextEditingController(text: session.notes);
+  final timeStartController = TextEditingController(text: timeToString(session.timeStart ?? TimeOfDay.now())!);
+  final timeEndController = TextEditingController(text: timeToString(session.timeEnd ?? TimeOfDay.now())!);
+  List<bool> selectedImpression = [false, false, false];
+  selectedImpression[session.impression - 1] = true;
+
+  showDialog(
```

Comment:
I had thought we could refactor `SessionPage` a bit, extract from there the form logic to its own form widget and simply reuse that here as well? Then we don't have two places that edit sessions